src/filter.c

```diff
@@ -974,7 +974,7 @@ int git_filter_list_stream_file(
 	}
 
 	if (readlen < 0)
-		error = readlen;
+		error = (readlen < INT_MIN) ? -1 : (int)readlen;
```

Comment:
POSIX specifies that `read` can only ever return `-1` or a value `>=0`. So you could just say:

```
if (readlen < 0)
    error = -1;
```

Reply:
Good call.  I had done this before I re-read the spec and been reminded that `ssize_t` is actually only guaranteed to be in the range of `(-1, SSIZE_MAX)`.  🤯

Reply:
Fixed.